### ISSUE TITLE
MIN_XXX_VERSION should take ENVIRONMENT into account

### ISSUE THREAD

**Issue by RReverser** [emscripten-core/emscripten#20561]
`MIN_XXX_VERSION` currently can drag supported feature set down even if `ENVIRONMENT` clearly excludes `XXX`.

Extreme example:

```
> ./emcc temp.cpp -o temp.js -s MIN_CHROME_VERSION=1 -s ENVIRONMENT=node
emcc: warning: enabling transpilation via closure due to browser version settings.  This warning can be suppressed by passing `--closure=1` or `--closure=0` to opt into our explicitly. [-Wtranspile]
```

In this case having `MIN_CHROME_VERSION` causes Emscripten to perform transpilation, even though target `ENVIRONMENT` is clearly specified to only care about Node.js.

Same can happen in the other direction with `-s ENVIRONMENT=web` being dragged down by `MIN_NODE_VERSION`, and it can happen even when user doesn't specify any custom version overrides, merely with built-in defaults.

As a more realistic example: our `MIN_NODE_VERSION` default is set to Node.js 16.0, so building with `-s ENVIRONMENT=node` should enable bulk memory by default, but currently doesn't because minimum Firefox version defaults to 68 and minimum Safari version defaults to 14.1, both of which didn't have bulk memory. So Node.js projects don't get bulk memory even though they don't care about those browsers. cc @sbc100 @kripken

**Comment by sbc100** [comment]
Yes, it would make sense to me to ignore `MIN_XX_VERSION` when its excluded by `ENVIRONMENT`.

In addition, we should probably warn error if `MIN_XXX_VERSION` is specified on the command line but XXX is unsupported by the specified `ENVIRONMENT`

**Comment by RReverser** [comment]
> In addition, we should probably warn error if `MIN_XXX_VERSION` is specified on the command line but XXX is unsupported by the specified `ENVIRONMENT`

Yup sounds reasonable.

I think ignoring it could be done in centralised place by automatically setting MIN_XXX_VERSION to "target not supported" for unsupported environments, so all the manual version checks would automatically do the right thing.
